src/nnet3/online-nnet3-decodable.cc

```diff
@@ -0,0 +1,210 @@
+// nnet3/online-nnet3-decodable.cc
+
+// Copyright  2014  Johns Hopkins University (author: Daniel Povey)
+//            2016  Api.ai (Author: Ilya Platonov)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nnet3/online-nnet3-decodable.h"
+#include "nnet3/nnet-utils.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+DecodableNnet3Online::DecodableNnet3Online(
```

Comment:
Actually I don't think that structure really necessary in this case, unless
there is some compelling reason for it (which I can't think of now).

BTW, the reason for Vimal's previous comment, regarding the "-simple" in
the name, is that we expect a certain structure of nnet, with inputs
"input" and possibly "ivector", and an output called "output".  This should
be checked using the IsSimpleNnet() function.
Dan


On Sun, Apr 10, 2016 at 5:33 PM, Vimal Manohar <notifications@github.com>
wrote:

> In src/nnet3/online-nnet3-decodable.cc
> <https://github.com/kaldi-asr/kaldi/pull/666#discussion_r59145085>:
>
> > +//  http://www.apache.org/licenses/LICENSE-2.0
> > +//
> > +// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
> > +// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
> > +// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
> > +// MERCHANTABLITY OR NON-INFRINGEMENT.
> > +// See the Apache 2 License for the specific language governing permissions and
> > +// limitations under the License.
> > +
> > +#include "nnet3/online-nnet3-decodable.h"
> > +#include "nnet3/nnet-utils.h"
> > +
> > +namespace kaldi {
> > +namespace nnet3 {
> > +
> > +DecodableNnet3Online::DecodableNnet3Online(
>
> To be similar to the code in
> https://github.com/api-ai/kaldi/blob/chain/src/nnet3/nnet-am-decodable-simple.cc,
> this class needs to be separated into a base class (e.g.
> OnlineNnet3DecodableBase) with only nnet and priors objects but not am_nnet
> and trans_model, and a child (e.g. OnlineNnet3DecodableAmNnetSimple) that
> has the am_nnet and trans_model objects. This might be required to create
> an nnet3 version of
> https://github.com/api-ai/kaldi/blob/75f8b027a2e6b6b679d26d0b4b667cea631ee6d1/src/online2bin/online2-wav-nnet2-am-compute.cc
> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/666/files/75f8b027a2e6b6b679d26d0b4b667cea631ee6d1#r59145085>
>